### ISSUE TITLE
DetailsList Column Header not announced for certain columns using Screen Reader

### ISSUE THREAD

**Issue by kfong52** [microsoft/fluentui#15365]
Navigating to a cell in a row should announce its column header

All columns have their column header announced here: https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/basic

Only some column headers are announced here: https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist

Was said to be resolved here but still have issues: https://github.com/microsoft/fluentui/issues/6019#issuecomment-445990096

#### Actual behavior:

Column header isn't announced for columns like date modified or modified by for second link.
Column header is announced when data is empty in those cells.

#### Expected behavior:

Column headers should be announced for it's respective cell.

**Comment by tomi-msft** [comment]
@kfong52 Thanks for submitting the issue! We'll look into this more.
